Framework/DataHandling/src/LoadDetectorInfo.cpp

```diff
@@ -274,7 +280,8 @@ void LoadDetectorInfo::loadFromIsisNXS(const std::string &filename) {
     double pressure = detInfo.pressures[i];
     double thickness = detInfo.thicknesses[i];
 
-    updateParameterMap(pmap, det, l2, theta, phi, delta, pressure, thickness);
+    updateParameterMap(pmap, detectorInfo.detector(index), l2, theta, phi,
```

Comment:
Mainly about the fact that ComponentIDs are attached to the detector. However, for `Instrument 2.0` I don't see much point in having a `Detector` class at all. This isn't something to do anything about now.